azure-pipelines/templates/test.yml

```diff
@@ -1,4 +1,24 @@
 steps:
+  # This gives the AzCodeE2ETests service connection access to this pipeline.
```

Comment:
These templates (including test.yml) are used by all of our extensions. I believe these steps will fail for most of the extensions that use them, but A. don't have the service connection setup and B. don't have tests that need a service connection.

We need a way to let extensions opt-in to these steps.

I don't have a recommended solution for this yet, but wanted to post this feedback so we can discuss.

Reply:
Is there an environment variable we could check to conditionally add these steps? That's one idea I have

Reply:
That's a good callout, Alex! That was one of my concerns too. We did talk about potentially adding a `useAzureFederatedIdentityCredentials` environment variable or something, but we haven't finalized this. I will say though, I don't think these two steps should make any pipeline fail perse? It'll just be annoying because they will need to be independently authorized to use the `AzCodeE2ETests` service connection. But beyond that, the environment variables will be propagated and nothing done with them 🤷🏽

Reply:
> It'll just be annoying because they will need to be independently authorized to use the `AzCodeE2ETests` service connection.

I guess I was thinking it was unreasonable to do this for every extension pipeline even though they won't be used. We aren't the only team using these templates. (ex: the Docker extension) I'm not sure we want to force everyone to add it before merging.

Reply:
Okay so I added the condition, but it looks like the pipeline still needs to be authenticated for the service connection regardless, which makes sense, since the condition could be a runtime variable. That said, it looks like these pipelines are only used on main in the nightly schedule, since it looks like our GitHub pipelines don't use these 1ES templates. In that case, does it make sense that these pipelines will all now need to be permitted to use the service connections? Bear in mind it's only a one time action